tests/model/test_model.py

```diff
@@ -888,24 +907,54 @@ def test_update_private_message(
                     "topic": "party",
                 },
                 "lets_party",
+                "party",
+                "stream",
+                "change_all",
                 True,
             ),
         ],
     )
     def test_update_stream_message(
-        self, mocker, model, response, return_value, req, old_topic, footer_updated
+        self,
+        mocker,
+        model,
+        response,
+        return_value,
+        req,
+        old_topic,
+        new_topic,
+        stream_name,
+        propagate_mode,
+        footer_updated,
     ):
         self.client.update_message = mocker.Mock(return_value=response)
         model.index["messages"][req["message_id"]]["subject"] = old_topic
+        req["topic"] = new_topic
+        model.index["messages"][req["message_id"]]["display_recipient"] = stream_name
+        req["propagate_mode"] = propagate_mode
+
+        recent_moved_msgs = ""
+        if propagate_mode == "change_all":
+            recent_moved_msgs = "all"
+        if propagate_mode == "change_later":
+            recent_moved_msgs = "some"
+        if propagate_mode == "change_one":
+            recent_moved_msgs = "one"
 
         result = model.update_stream_message(**req)
-
         self.client.update_message.assert_called_once_with(req)
         assert result == return_value
         self.display_error_if_present.assert_called_once_with(response, self.controller)
         report_success = model.controller.report_success
         if result and footer_updated:
-            report_success.assert_called_once_with("You changed a message's topic.")
+            if recent_moved_msgs == "one":
```

Comment:
By just using a string we can assert on an 'expected string' provided in each parametrize 'case', avoiding the need for extra conditionals here.